cpukit/include/rtems/score/memorymanagement.h

```diff
@@ -0,0 +1,25 @@
+#ifndef _RTEMS_SCORE_MEMORYMANAGEMENT_H
+#define _RTEMS_SCORE_MEMORYMANAGEMENT_H
+
+#include <stdlib.h>
+#include <stdint.h>
+
+typedef enum
+{
+    READ_WRITE,
+    READ_WRITE_CACHED,
+    READ_ONLY,
+    READ_ONLY_CACHED,
+    NO_ACCESS
+} memory_flags;
+
+void memory_entries_set(uint32_t *begin_addr, size_t size, memory_flags flags);
```

Comment:
That's shared code. Assuming addresses are pointers to `uint32_t` isn't correct for 64-bit architectures. Please make it `uintptr_t`